cli/kubectl-hyperkube

```diff
@@ -17,14 +54,55 @@ def hkc_print(message, *args, **kwargs):
 @click.group()
 @click.option('--config', '-c', required=False,
               default=str(Path.home()) + "/hyperkube-config.yaml")
+@click.option('--hyperkube-url', required=False, default=None)
+@click.option('--hyperkube-stage', required=False, default=None)
+@click.option('--aws-profile', required=False, default=None)
+@click.option('--x-api-key', required=False, default=None)
+@click.option('--aws-region', required=False, default=None)
 @click.pass_context
-def cli(ctx, config):
+def cli(ctx, config, hyperkube_url,
+        hyperkube_stage, aws_profile, x_api_key, aws_region):
     ctx.ensure_object(dict)
+
+    # Load user provided config file, these options can be
+    # overidden by commandline flags
     ctx.obj['hyperkube_config'] = _load_config(config)
-    ctx.obj['url'] = ctx.obj['hyperkube_config']['url']
-    ctx.obj['stage'] = ctx.obj['hyperkube_config']['stage']
-    headers = {'X-Api-Key': ctx.obj['hyperkube_config']['x_api_key']}
-    ctx.obj['headers'] = headers
+
+    if hyperkube_url is None:
+        hyperkube_url = ctx.obj['hyperkube_config']['url']
+
+    u = urlparse(hyperkube_url)
+    hostname = u.netloc
+
+    if hyperkube_stage is None:
+        hyperkube_stage = ctx.obj['hyperkube_config']['stage']
+
+    if aws_region is None:
+        aws_region = ctx.obj['hyperkube_config'].get(
+                             'aws_region', DEFAULT_AWS_REGION)
+
+    if aws_profile is None:
+        aws_profile = ctx.obj['hyperkube_config'].get(
+                              'aws_profile', AWS_PROFILE)
+
+    auth = AWSProfileAuth(aws_host=hostname,
+                          aws_region=aws_region,
+                          aws_service='execute-api',
+                          profile=aws_profile)
+
+    ctx.obj['base_url'] = f'{hyperkube_url}/{hyperkube_stage}'
+    ctx.obj['auth'] = auth
+
+    # Load api key from cli flag. If not provided try config
+    if x_api_key is None:
```

Comment:
What about the following?
```
if x_api_key is None:
    x_api_key = ctx.obj['hyperkube_config'].get(
                'x_api_key')
if x_api_key is not None:
    ctx.obj['headers'] = {'X-Api-Key': x_api_key}
```

Reply:
I need to define what the header key is 'X-Api-Key' but i think I have something similar.

Reply:
How about:

```
    ctx.obj['headers'] = None
    if x_api_key is None and config:
        if ctx.obj['hyperkube_config'].get('x_api_key'):
            headers = {'X-Api-Key': ctx.obj['hyperkube_config'].get(
                'x_api_key')}
            ctx.obj['headers'] = headers
    if x_api_key is not None:
        ctx.obj['headers'] = {'X-Api-Key': x_api_key}
```